src/Illuminate/Support/ServiceProvider.php

```diff
@@ -77,12 +89,23 @@ protected function loadRoutesFrom($path)
      *
      * @param  string  $path
      * @param  string  $namespace
+     * @param  bool  $publish
      * @return void
      */
-    protected function loadViewsFrom($path, $namespace)
+    protected function loadViewsFrom($path, $namespace, $publish = false)
     {
-        if (is_dir($appPath = $this->app->resourcePath().'/views/vendor/'.$namespace)) {
-            $this->app['view']->addNamespace($namespace, $appPath);
+        if (! $this->viewNamespacesRegistered) {
```

Comment:
@taylorotwell  FYI—I added this check because if someone *were* to call `loadViewsFrom` multiple times as you suggested, it would re-register the default namespace each time.